docs/qupath.rst

```diff
@@ -5,99 +5,141 @@ Description
+#. You can adjust the parameters. Click ``Run``. This will draw red ROIs around cells and nuclei inside your ``Annotation``.
 
    |image4|
 
-#. Select ``Measure > Show detection measurements``.
+#. Click on ``Hierarchy`` tab in the left-hand pane of QuPath. Expand the ``Annotation`` you have just run the ``Cell detection`` on.
+
+#. Select several detection ROIs.
+
+#. Open the scripting dialog in QuPath ``Automate > Show script editor`` and paste into it the following code::
 
-   |image5|
+      import qupath.lib.images.servers.omero.OmeroTools
+      OmeroTools.writePathObjects(getSelectedObjects(), getCurrentServer())
 
-#. Note: You can save the results locally by clicking ``Save`` in the bottom right of the ``Detection results table``. If you are using your own server, you can upload the results and link them to the Image.
+#. From the top menu, select ``Run > Run``. This saves the detection ROIs you selected in the ``Hierarchy`` tab into OMERO.
 
-#. In the following steps, we will show how to convert the ROIs your just created in QuPath into OMERO ROIs and attach them to the image in OMERO.
+#. Go to OMERO.iviewer and refresh the image. Inspect the saved detection ROIs.
 
-#. First, use the ROI OME-XML export script to export your ROIs from QuPath into OME-XML file. Find the version of ``ome-omero-roitool`` mentioned in Resources on `ome-omero-roitool releases <https://github.com/glencoesoftware/ome-omero-roitool/releases>`_ and from there download the ``ome-omero-roitool-xxx.zip``. The downloaded zip contains both the plugin and the QuPath scripts needed for this workflow.
+.. _Roitool:
+
+Save detection ROIs using ome-omero-roitool
+-------------------------------------------
+This workflow necessritates the usage of the Command Line Interface. The limitation here are the Annotation ROIs, whcih are transformed into masks in OMERO. Although this preserves the holes in the Annotations, if the Annotation ROIs are too large, it might result in performance problems or even running out of resources on the machine where the export of the mask from QuPath is attempted.
+
+#. Connect QuPath to OMERO, open an image from OMERO in QuPath and draw an ``Annotation`` on it as described in :ref:`Opening images with ROIs from OMERO in QuPath<OpeninginQuPath>`.
+
+#. Adjust your ``Annotation`` using the ``Brush`` tool |image3|.
+
+#. Select ``Analyze > Cell detection > Cell detection``.
+
+#. You can adjust the parameters. Click ``Run``. This will draw red ROIs around cells and nuclei inside your ``Annotation``.
+
+#. Use the ROI OME-XML export script to export your ROIs from QuPath into OME-XML file. Find the version of ``ome-omero-roitool`` mentioned in Resources on `ome-omero-roitool releases <https://github.com/glencoesoftware/ome-omero-roitool/releases>`_ and from there download the ``ome-omero-roitool-xxx.zip``. The downloaded zip contains both the plugin and the QuPath scripts needed for this workflow.
```

Comment:
Now that you have this, I think you can remove the duplicated bits from steps below: "If you did not do so already, find the version of the ome-omero-roitool mentioned in Resources on ome-omero-roitool releases. From there, download the ome-omero-roitool-xxx.zip" and "Unzip the downloaded file" and even `unzip ome-omero-roitool-xxx.zip`.

Reply:
fixed in https://github.com/ome/omero-guide-qupath/pull/6/commits/00853dd5d6c8135e1f1e9258864952c96da462e1